onnxruntime/core/session/IOBinding.h

```diff
@@ -100,8 +100,10 @@ class IOBinding {
   IOBinding(const SessionState& session_state);
   const SessionState& session_state_;
   std::vector<std::string> feed_names_;
+  std::unordered_map<std::string, size_t> mapped_feed_names_;
```

Comment:
>std::unordered_map<std::string, [](http://example.com/codeflow?start=2&length=31)

Can you, please, store reference_wrappers pointing to the actual strings? You will have to specify hash and compare function with <std::string> manually

Reply:
The duplicated strings are removed. The tree (in std::map) is now modified inplaced to avoid a new search. Maybe there is a better implementation. A new container was created to access protected methods of std::map.